jsonnet/prometheus-adapter.libsonnet

```diff
@@ -94,6 +94,7 @@ function(params)
                           '--metrics-relist-interval=1m',
                           '--prometheus-url=' + cfg.prometheusURL,
                           '--secure-port=6443',
+                          '--tls-cipher-suites=TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256,TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_256_CBC_SHA',
```

Comment:
We should have one set of ciphersuites for all components and not store them in different places.

This list should either be taken from any kube-rbac-proxy sidecar settings. Or it should be configured in `main.jsonnet` and propagated everywhere from there.

/hold

Reply:
note: `kube-rbac-proxy` component allows easy configuration of ciphersuites in https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/components/kube-rbac-proxy.libsonnet#L11-L38. We might want to surface that in components that use KRP.

Reply:
We already have an epic for 4.10 to configure TLS settings from a central OpenShift API. I changed the PR to take the ciphers from the thanos querier: https://github.com/openshift/cluster-monitoring-operator/blob/master/jsonnet/thanos-querier.libsonnet#L450

Reply:
> We already have an epic for 4.10 to configure TLS settings from a central OpenShift API, so I wouldn't try to solve that problem now. 

This is a different problem. Applying any configuration from a central API is a runtime issue, jsonnet code and manifest generation is done before runtime.

> I changed the PR to take the ciphers from the thanos querier

This is taking current ciphers and do not allow any simple changes. As such it is a stop-gap and not a proper solution.

Reply:
I am not sure if this is really necessary. I would expect the static tls-cipher-suites to be removed as an argument from static jsonnet files once we move from configuring them at runtime. In any case, I've added a variable in main.jsonnet and propagated it downstream to all components. I would also be mindful of the scope of the Bugzilla this PR addresses.

Reply:
> I would expect the static tls-cipher-suites to be removed as an argument from static jsonnet files once we move from configuring them at runtime.

Those assumptions may be valid for 4.10 and we are in 4.9 development cycle. This OpenShift version will need to be supported for ~1 year or more. During that time there is no guarantee we won't need to change cipher suites for all components and if that is the case I would prefer to change it in one place instead of searching other implementations.

Reply:
Ah ok that makes sense. I didn't take into account the longer support cycle :+1: